CONTRIBUTING.md

```diff
@@ -54,36 +194,78 @@ questions, and
 [#Node-dev](http://webchat.freenode.net/?channels=node-dev) for development of
 Node.js core specifically.
 
-### Step 2: Branch
+### Setting up your local environment
+
+To get started, you will need to have `git` installed locally. Depending on
+your operating system, there are also a number of other dependencies required.
+These are detailed in the [Building guide][].
+
+Once you have `git` and are sure you have all of the necessary dependencies,
+it's time to create a fork.
 
-Create a branch and start hacking:
+Before getting started, it is recommend to configure `git` so that it knows
+who you are:
 
 ```text
-$ git checkout -b my-branch -t origin/master
+$ git config --global user.name "J. Random User"
+$ git config --global user.email "j.random.user@example.com"
 ```
 
-Any text you write should follow the [Style Guide](doc/STYLE_GUIDE.md),
-including comments and API documentation.
+#### Step 1: Fork
+
+Fork the project [on GitHub](https://github.com/nodejs/node) and clone your fork
+locally.
+
+```text
+$ git clone git@github.com:username/node.git
+$ cd node
+$ git remote add upstream https://github.com/nodejs/node.git
+$ git fetch upstream
+```
 
-### Step 3: Commit
+#### Step 2: Branch
 
-Make sure git knows your name and email address:
+As a best practice to keep your development environment as organized as
+possible, create local branches to work within. These should also be created
+directly off of the `master` branch.
 
 ```text
-$ git config --global user.name "J. Random User"
-$ git config --global user.email "j.random.user@example.com"
+$ git checkout -b my-branch -t origin/master
 ```
 
-Add and commit:
+### The Process of Making Changes
+
+#### Step 3: Code
+
+The vast majority of pull requests opened against the `nodejs/node`
+repository includes changes to either the C/C++ code contained in the `src`
+directory, the JavaScript code contained in the `lib` directory, the
+documentation in `docs/api` or tests within the `test` directory.
+
+If you are modifying code, please be sure to run `make lint` from time to
+time to ensure that the changes follow the Node.js code style guide.
+
+Any documentation you write (including code comments and API documentation)
+should follow the [Style Guide](doc/STYLE_GUIDE.md).
```

Comment:
Should we mention that code fragments in docs are also the subject for the `make lint`?